src/main/scala/is/hail/variant/ReferenceGenome.scala

```diff
@@ -443,9 +444,43 @@ case class ReferenceGenome(name: String, contigs: Array[String], lengths: Map[St
         contigs.map(contig => JSONExtractContig(contig, contigLength(contig))),
         xContigs, yContigs, mtContigs,
         par.map(i => JSONExtractIntervalLocus(i.start.asInstanceOf[Locus], i.end.asInstanceOf[Locus])))
-      implicit val formats = defaultJSONFormats
+      implicit val formats: Formats = defaultJSONFormats
       Serialization.write(jrg, out)
     }
+
+  def toJSONString: String = {
+    val jrg = JSONExtractReferenceGenome(name,
+      contigs.map(contig => JSONExtractContig(contig, contigLength(contig))),
+      xContigs, yContigs, mtContigs,
+      par.map(i => JSONExtractIntervalLocus(i.start.asInstanceOf[Locus], i.end.asInstanceOf[Locus])))
+    implicit val formats: Formats = defaultJSONFormats
+    Serialization.write(jrg)
+  }
+
+  def addAsField(fb: FunctionBuilder[_]): (ClassFieldRef[ReferenceGenome], Code[Unit]) = {
+    val json = toJSONString
+    val chunkSize = 65535
+    val nChunks = (json.length() - 1) / chunkSize + 1
+    assert(nChunks > 0)
+    val stringRef = fb.newField[String](s"strrg_$name")
```

Comment:
I'm not sure there's a guarantee that RGs have no spaces. Would that cause a problem here?